puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman_packaging_release.groovy

```diff
@@ -34,7 +34,11 @@ pipeline {
 
                 script {
 
-                    if (fileExists('commit')) {
+                    if (params.previous_commit) {
```

Comment:
I am not entirely sure this condition is correct, tho.

Reply:
Maybe it needs to check if it's a non-empty string? I've learned to be defensive about that.

Reply:
I was thinking this could be simpler by dropping the `else` clause that builds all packages. And instead, if no commit file exists take the current commit of HEAD of the packaging repository. The only time this commit file should be lacking is when a new job is created and that would take the commit at the time of the initial run of the job.

Reply:
I like that idea. I was thinking that you'd want to do this on job creation but just taking HEAD is much more elegant. We should make sure to update the procedure to call the job right after branching so we don't miss a commit.